gridformat/common/hdf5.hpp

```diff
@@ -502,9 +505,15 @@ class File {
     auto _read_buffer_field(const Source& source) const {
         auto dims = source.getMemSpace().getDimensions();
         MDLayout layout = dims.empty() ? MDLayout{{1}} : MDLayout{std::move(dims)};
-        std::vector<T> out(layout.number_of_entries());
+
+        using _T = Detail::HighFiveDataType<T>;
+        std::vector<_T> out(layout.number_of_entries());
         source.read(out.data());
-        return BufferField(std::move(out), std::move(layout));
+
+        if constexpr (std::is_same_v<T, _T>)
+            return BufferField(std::move(out), std::move(layout));
+        else
+            return BufferField(std::vector<T>{out.begin(), out.end()}, std::move(layout));
```

Comment:
necessary? I think `BufferField(std::move(out), std::move(layout));` compiled for me with bool and char

Reply:
I thought that if `T` is set to be `bool`, that means the user wants a `bool` field so I cast it back. But let me double check the logic where T comes from.

Reply:
But on compiler explorer this does not seem to compile (which does not surprise me), or am I missing something here?

```cpp
#include <vector>
int main() {
    std::vector<char> a{'a', 'b'};
    std::vector<bool> v{std::move(a)};
    return 0;
}
```

Reply:
ah yeah you forward the buffer into an explicitly instantiated `BufferField<S>`, so the actual seiralization (and copy) happens in there. That's nicer! Can I squash your commit into the other and force-push?

Reply:
ok